rs/orchestrator/src/registration.rs

```diff
@@ -664,14 +665,21 @@ fn process_ipv4_config(
 
 fn process_domain_name(log: &ReplicaLogger, domain: &str) -> OrchestratorResult<Option<String>> {
     info!(log, "Reading domain name for registration");
-    match domain {
-        "" => Ok(None),
-        domain if is_valid_domain(domain) => Ok(Some(domain.into())),
-        _ => Err(OrchestratorError::invalid_configuration_error(format!(
-            "Provided domain name {} is invalid",
-            domain
-        ))),
+    if domain.is_empty() {
+        return Ok(None);
     }
+
+    if domain_to_ascii_strict(domain)
+        .ok()
+        .filter(|s| s == domain)
+        .is_none()
```

Comment:
nit: Is this equivalent to your version? If so, I slightly prefer this version
```suggestion
    if !domain_to_ascii_strict(domain).is_ok_and(|s| s == domain)
```

Reply:
nice I didn't know this